src/hive_mind_bridge/src/main.cpp

```diff
@@ -86,12 +94,17 @@ int main(int argc, char** argv) {
                                           int argsLength) -> std::optional<CallbackReturn> {
         swarmus_ros_navigation::MoveByMessage moveByMessage;
 
-        moveByMessage.distance_x = std::get<float>(args[0].getArgument());
-        moveByMessage.distance_y = std::get<float>(args[1].getArgument());
+        std::visit(
+            Overload{[&](auto&& arg) { moveByMessage.distance_x = arg; }, [](std::monostate) {}},
+            args[0].getArgument());
+
+        std::visit(
+            Overload{[&](auto&& arg) { moveByMessage.distance_y = arg; }, [](std::monostate) {}},
+            args[1].getArgument());
```

Comment:
Seems very complicated to read. I wouldn't put this in the documentation, but I guess it's a good way to make the code safer.

Reply:
I know it's a bit clunky, but idk how to make it readable without having too many classes. 

Would something like this be easier to read?

`
Overload{
   [&](int& arg) { moveByMessage.distance_y = arg; }, 
   [&](float& arg) { moveByMessage.distance_y = arg; }, 
   [](std::monostate) {}}
`

We can't use auto since the monostate cannot be casted to a string

Reply:
So this code aims to handle both ints and floats? I don't think we should do that. The manifest that is declared just after specifically says that moveBy should take `float x` and `float y` as arguments. 

I intended to force this strict typing in the design. To be coherent, we should produce an error when a wrong type is passed. So that way, the simplest way to acquire the input arguments would be to surround this with a `std::get_if`, and log an error if there is a type error.

What dou you think?